code/game/machinery/computer/abnormality_archive.dm

```diff
@@ -0,0 +1,89 @@
+			var/obj/item/paper/fluff/info/record_detail = _record_detail
+			var/list/record_data = list()
+			if(initial(record_detail.name) == "paper" || initial(record_detail.noArchive) == TRUE) //removes template papers and allows for secret files
+				continue
+			note_list = splittext(initial(record_detail.info), "<br>") //splits text into a list by making <br> into a break,
+			popleft(note_list) //Remove first line since it is the name
+			pop(note_list) //Removes last line since it is almost always blank. If you are making your abnormality and encountering this issue i give you my blessing to remove this line.
+			linecount = note_list.len //Doing it seperately results in only half of the lines being given a value. A single variable seems to keep it stable.
+			if(!linecount)
+				return
+			record_data["linecount"] = linecount //len is list length
+			record_data["type"] = record_detail //gives the typepath of the file we are reading
+			record_data["name"] = remove_paper_commands(initial(record_detail.name))
+			for(var/i=0,i <= linecount,i++) //loop until we reach the same length as the linecount
+				record_data["line[i]"] += remove_paper_commands(popleft(note_list))
+			abnormality_info += list(record_data)
+
+	.["abnormality_info"] = abnormality_info
+
+/obj/machinery/computer/abnormality_archive/ui_act(action, params) //Basically catches signals caused by the Javascript act('print_file', { ref: currentAbnormality.type })
+	. = ..()
+	if(.)
+		return
+	switch(action)
+		if("print_file")
+			if(paper <= 0)
+				return
+			printfile = params["ref"]
+			print_file(printfile)
+
+/obj/machinery/computer/abnormality_archive/attackby(obj/item/O, mob/user)
+	if(istype(O, /obj/item/paper) && paper <= papermax)
+		qdel(O)
+		paper = paper + 1
+	..()
+
+/obj/machinery/computer/abnormality_archive/proc/print_file(file)
+	paper = paper - 1
+	visible_message("<span class='notice'>[src] begins to print a file.</span>")
+	playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 30, TRUE)
+	new file(get_turf(src))
+
+/obj/machinery/computer/abnormality_archive/proc/remove_paper_commands(paper) //Removes all command code. WILL cause problems for abnormality documents later. I am sorry about this.
+	note = replacetext(paper, "<br>", "")
+	note = replacetext(note, "<li>", "")
+	note = replacetext(note, "'", "")
+	note = replacetext(note, "<ul>", "")
+	note = replacetext(note, "</ul>", "")
+	note = replacetext(note, "<center>", "")
+	note = replacetext(note, "</center>", "")
+	note = replacetext(note, "<strong>", "")
+	note = replacetext(note, "</strong>", "")
+	note = replacetext(note, "<h1>", "")
+	note = replacetext(note, "</h1>", "")
+	note = replacetext(note, "<h2>", "")
+	note = replacetext(note, "</h2>", "")
+	note = replacetext(note, "<h3>", "")
+	note = replacetext(note, "</h3>", "")
+	note = replacetext(note, "<h4>", "")
+	note = replacetext(note, "</h4>", "")
+	note = html_encode(note)
+	return note
```

Comment:
I have a feeling there are already established sanitized procs for this like BYONDs native `html_encode` and `sanitize_simple` and the wrapper `sanitize`, all in `code->_HELPERS->text.dm`. Perhaps use those.

Reply:
Reason for why I'm directing you to them:
A) Don't reinvent the wheel
B) Sanitization is very important to get right - if anything breaks, we want to have to update as few places as possible as fast as possible

Reply:
When i use sanitize(paper) it returns 
![image](https://user-images.githubusercontent.com/109536843/208725209-94d12811-fde0-44c4-83c1-137bd4b0051d.png)

Reply:
Also PDA seems to use similar code. when it scans documents.

Reply:
if (!scanmode && istype(A, /obj/item/paper) && owner)
		var/obj/item/paper/PP = A
		if (!PP.info)
			to_chat(user, "<span class='warning'>Unable to scan! Paper is blank.</span>")
			return
		notehtml = PP.info
		note = replacetext(notehtml, "<BR>", "\[br\]")
		note = replacetext(note, "<li>", "\[*\]")
		note = replacetext(note, "<ul>", "\[list\]")
		note = replacetext(note, "</ul>", "\[/list\]")
		note = html_encode(note)
		notescanned = TRUE
		to_chat(user, "<span class='notice'>Paper scanned. Saved to PDA's notekeeper.</span>" )